horovod/spark/keras/remote.py

```diff
@@ -236,56 +225,42 @@ def train(serialized_model, train_rows, val_rows, avg_row_size):
-                is_batch_reader = True
-
-            with reader_factory(remote_store.train_data_path,
-                                num_epochs=1,
-                                cur_shard=hvd.rank(),
-                                reader_pool_type=reader_pool_type,
-                                workers_count=train_reader_worker_count,
-                                shard_count=hvd.size(),
-                                hdfs_driver=PETASTORM_HDFS_DRIVER,
-                                schema_fields=schema_fields,
-                                transform_spec=transform_spec,
-                                storage_options=storage_options,
-                                shuffle_rows=shuffle,
-                                shuffle_row_groups=shuffle,
-                                seed=random_seed,
-                                **reader_factory_kwargs) as train_reader:
-                with reader_factory(remote_store.val_data_path,
-                                    num_epochs=1,
-                                    cur_shard=hvd.rank(),
-                                    reader_pool_type=reader_pool_type,
-                                    workers_count=val_reader_worker_count,
-                                    shard_count=hvd.size(),
-                                    hdfs_driver=PETASTORM_HDFS_DRIVER,
-                                    schema_fields=schema_fields,
-                                    transform_spec=transform_spec,
-                                    storage_options=storage_options,
-                                    shuffle_rows=False,
-                                    shuffle_row_groups=False,
-                                    **reader_factory_kwargs) \
-                    if should_validate else empty_batch_reader() as val_reader:
-
-                    train_data = make_dataset(train_reader, batch_size,
-                                              is_batch_reader, shuffle=shuffle, cache=inmemory_cache_all)
-                    val_data = make_dataset(val_reader, val_batch_size,
-                                            is_batch_reader, shuffle=False, cache=inmemory_cache_all) \
-                        if val_reader else None
-
-                    history = fit(model, train_data, val_data, steps_per_epoch,
-                                  validation_steps, callbacks, verbose)
+
+            data_module_kwargs = {
+                'label_cols': label_columns,                    # nvtabular
+                'continuous_cols': continuous_columns,          # nvtabular
+                'categorical_cols': categorical_columns,        # nvtabular
+                'make_dataset': make_dataset,                   # petastorm
+                'random_seed': random_seed,                     # petastorm
+                'train_dir': remote_store.train_data_path,
+                'val_dir': remote_store.val_data_path,
+                'num_train_epochs': epochs,
```

Comment:
It is okay to pass `num_train_epochs` for generic `DataModule` constructor, though it is not used in either Petastorm or NVTabular. But I think `number of epochs` is a generic attribute of data loader.